tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml

```diff
@@ -18,7 +18,7 @@ spec:
           katacontainers.io/kata-runtime: cleanup
       containers:
       - name: kube-kata-cleanup
-        image: quay.io/kata-containers/kata-deploy:latest
+        image: quay.io/kata-containers/kata-deploy:3.0.0-rc1
```

Comment:
Looks like we should not change this by hand. Instead, the `test-kata.sh` script changes it automatically for stable releases. While for non-stable releases, we are using `latest` here. Are the diffs generated by script?

Reply:
I think that's why https://github.com/kata-containers/kata-containers/actions/runs/3142238852 fails, because we do not have `:3.0.0-rc1` kata-deploy image yet (which is created when the new tag is pushed).

Reply:
> Looks like we should not change this by hand. Instead, the `test-kata.sh` script changes it automatically for stable releases. While for non-stable releases, we are using `latest` here. Are the diffs generated by script?

Yes all the changes in this PR were generated by the following scriplet without any further action from my part:
```
$ export NEW_VERSION=3.0.0-rc1
$ export BRANCH=stable-3.0
$ ./update-repository-version.sh -p "$NEW_VERSION" "$BRANCH"
```

Reply:
> I think that's why https://github.com/kata-containers/kata-containers/actions/runs/3142238852 fails, because we do not have `:3.0.0-rc1` kata-deploy image yet (which is created when the new tag is pushed).

It seems you're right but how did we get there ?!?

Reply:
This is supposedly fixed by https://github.com/kata-containers/kata-containers/pull/5284